lib/sycamore/sycamore/transforms/text_extraction/pdf_miner.py

```diff
@@ -0,0 +1,108 @@
+from sycamore.data import Element, BoundingBox
+from sycamore.utils.cache import DiskCache
+from typing import BinaryIO, Tuple, List, cast, Generator, TYPE_CHECKING
+from pathlib import Path
+from sycamore.utils.import_utils import requires_modules
+import logging
+
+if TYPE_CHECKING:
+    from pdfminer.layout import LTPage
+
+logger = logging.getLogger(__name__)
+
+pdf_miner_cache = DiskCache(str(Path.home() / ".sycamore/PDFMinerCache"))
+
+
+class PDFMinerExtractor:
+    @requires_modules(["pdfminer", "pdfminer.utils"], extra="local-inference")
+    def __init__(self):
+        from pdfminer.converter import PDFPageAggregator
+        from pdfminer.layout import LAParams
+        from pdfminer.pdfinterp import PDFPageInterpreter, PDFResourceManager
+
+        rm = PDFResourceManager()
+        param = LAParams()
+        self.device = PDFPageAggregator(rm, laparams=param)
+        self.interpreter = PDFPageInterpreter(rm, self.device)
+
+    @staticmethod
+    @requires_modules(["pdfminer", "pdfminer.utils"], extra="local-inference")
+    def pdf_to_pages(file_name: str) -> Generator["LTPage", None, None]:
+        from pdfminer.converter import PDFPageAggregator
+        from pdfminer.layout import LAParams
+        from pdfminer.pdfinterp import PDFPageInterpreter, PDFResourceManager
+        from pdfminer.pdfpage import PDFPage
+        from pdfminer.utils import open_filename
+
+        rm = PDFResourceManager()
+        param = LAParams()
+        device = PDFPageAggregator(rm, laparams=param)
+        interpreter = PDFPageInterpreter(rm, device)
+        with open_filename(file_name, "rb") as fp:
+            fp = cast(BinaryIO, fp)
+            pages = PDFPage.get_pages(fp)
+            for page in pages:
+                interpreter.process_page(page)
+                page_layout = device.get_result()
+                yield page_layout
+
+    @staticmethod
+    def _convert_bbox_coordinates(
+        rect: Tuple[float, float, float, float],
+        height: float,
+    ) -> Tuple[float, float, float, float]:
+        """
+        pdf coordinates are different, bottom left is origin, also two diagonal points defining a rectangle is
+        (bottom left, upper right), for details, refer
+        https://www.leadtools.com/help/leadtools/v19/dh/to/pdf-topics-pdfcoordinatesystem.html
+        """
+        x1, y2, x2, y1 = rect
+        y1 = height - y1
+        y2 = height - y2
+        return x1, y1, x2, y2
+
+    def extract_document(self, filename: str, hash_key: str, use_cache=False, **kwargs) -> List[List[Element]]:
```

Comment:
Can you add a TODO to clean this up once we have the implementation in the service?

Reply:
Done.